src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -324,6 +324,31 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  override def getBucketUsage(bucketName: String, maxResults: Option[Long]): Future[BigInt] = {
+    def sumItemSizes(items: java.util.List[StorageObject]) = {
+      (BigInt(0) /: items) { (sum, o) => sum + o.getSize }
+    }
+
+    def processPage(pageToken: Option[String] = None): Future[BigInt] = {
+      val fetcher = getStorage(getBucketServiceAccountCredential).objects().list(bucketName)
+      if (maxResults.isDefined) fetcher setMaxResults maxResults.get
+      if (pageToken.isDefined) fetcher setPageToken pageToken.get
+
+      retryWithRecoverWhen500orGoogleError(() => {
+        val result = executeGoogleRequest(fetcher)
+        result match {
+          case r if r.getItems == null => BigInt(0)
+          case r if r.getNextPageToken == null => sumItemSizes(r.getItems)
+          case r => sumItemSizes(r.getItems) + Await.result(processPage(Some(result.getNextPageToken)), Duration.Inf)
+        }
+      }) {
+        case e: HttpResponseException => throw new RawlsException(s"Error gathering storage bucket usage: ${e.getMessage}")
+      }
+    }
+
+    processPage()
```

Comment:
I was getting an error without the () because of the one optional parameter. I agree, I wish I could remove the (). I'm also not entirely satisfied with the code. I could remove the default parameter value and explicitly pass in None here, but then I don't like how that reads. I may see if I can make it more clear by splitting out fetchFirstPage() and fetchNextPage(String) functions.